src/main/java/sparksoniq/jsoniq/compiler/translator/expr/flowr/FlworClause.java

```diff
@@ -50,8 +54,9 @@ public void setPreviousClause(FlworClause previousClause) {
         this.previousClause = previousClause;
     }
 
-    public FLWOR_CLAUSES getClauseType() {
-        return clauseType;
+    @Override
+    public void initHighestExecutionMode() {
+        this._highestExecutionMode = previousClause.getHighestExecutionMode();
```

Comment:
Could we add a comment that this is meant to be overridden in some cases? Thanks!

Reply:
Added

Reply:
Thanks!